### ISSUE TITLE
Batching analytics requests from multiple tag

### ISSUE THREAD

**Issue by zhouyx** [ampproject/amphtml#21727]
amp-analytics supports batching analytics requests today, but requests from multiple `<amp-analytics>` components don't get batched. 

This can be an issue when trying to track element that's dynamic loaded using `<amp-list>` and template. For example, 

```
<amp-list ....>
  <template type=....>
    <amp-analytics type=xxxx>
     </amp-analytics>
  </template>
</amp-list>
```

It is current difficult to support batching requests from multiple analytics tag, because all requests are scoped within the component itself. I want to understand how common the use case is and get everyone's idea on this. Thank you

cc @jeffjose @cathyxz @lannka

**Comment by cathyxz** [comment]
Is the reason for multiple analytics tag just because they want to track the visibility of each element in an `<amp-list>`? Aside from `visibility`, what are things that they might want to track? Would it be reasonable to add some kind of logic in `<amp-list>` to fire certain analytics events (e.g. visibility, click) per item? Does this apply to vanilla `<amp-list>` or just infinite scroll / load-more? 

/cc @nainar re: scoping analytics in `<amp-list> v2`.

**Comment by cathyxz** [comment]
We chatted about this offline--to summarize: 

It seems like the best option forward is as follows: 

- `<amp-analytics>` has a per instance `querySelector` that currently only selects 1 item. We can expand this to select multiple items so that it can track all items on `<amp-list>`. 
- `<amp-list>` can change / update its items via a bound `[src]` or `load-more`. This triggers the `DOM_UPDATE` event, and we can have `<amp-analytics>` rescan on `DOM_UPDATE`. 

Implementation details to consider are: 
- Whether to allow generic querySelector, or just a whitelist allowing children of amp-list only, or to even add an attribute for tracked items so that it's difficult for people to accidentally write an incorrect query selector and end up tracking items they did not intend to track. 

Other options considered include: 
- Having `<amp-list>` expose a function to return all of its current children. 
- Having `<amp-analytics>` expose a rescan function that is called by `<amp-list>`. 

Basically I think we would like to make this as generic as possible and reduce interdependency between `<amp-analytics>` and `<amp-list>`, hence the first option here is the preferred one. If that doesn't work, we can explore other options. =)
